strawberry_django/optimizer.py

```diff
@@ -473,18 +470,30 @@ def _get_model_hints(
     if pk is not None:
         store.only.append(pk.attname)
 
-    for f_selection in get_selections(selection, typename=t_name).values():
-        field = fields.get(f_selection.name, None)
+    for f_selections in _get_selections(info, parent_type).values():
+        f_selection = f_selections[0]
+        field = fields.get(f_selection.name.value, None)
         if not field:
             continue
 
         # Do not optimize the field if the user asked not to
         if getattr(field, "disable_optimization", False):
             continue
 
+        field_definition = parent_type.fields[f_selection.name.value].type
+        while isinstance(field_definition, GraphQLWrappingType):
+            field_definition = field_definition.of_type
+
+        f_info = _generate_selection_resolve_info(
+            info,
+            f_selections,
+            field_definition,
+            parent_type,
+        )
+
         # Add annotations from the field if they exist
         field_store = getattr(field, "store", None)
-        if field_store is not None:
+        if field_store:
```

Comment:
the store has a `__bool__` method that will return `False` if it is empty, which avoids the overhead of doing the `|=` below